Sources/DulocCastle/ASN1/Asn1ContentMetdata.swift

```diff
@@ -0,0 +1,283 @@
+//
+//  Asn1ContentMetdata.swift
+//  
+//
+//  Created by Kevin Miller on 7/23/22.
+//
+
+import Foundation
+
+enum Asn1MetadataBitmask: UInt8 {
+    // Initial Byte
+    case CLASS    = 0b11000000
+    case METHOD   = 0b00100000
+    case TAG      = 0b00011111
+    
+    // Length Parsing
+    
+    // Modifier bit actions
+    // Tag Len      => Short/Long form
+    // Tag Len      => If Long form, continue if present
+    // Content Len  => Short/Long form
+    case LEN_MODIFIER   = 0b10000000
+    case LEN_CONTENT    = 0b01111111
+    case LEN_MAX_WRITE  = 0b01111110
+}
+
+enum Asn1IdClass: UInt8 {
+    case UNIVERSAL      = 0x00
+    case APPLICATION    = 0x01
+    case CONTEXTUAL     = 0x02
+    case PRIVATE        = 0x03
+    
+    init(leadingByte: UInt8) {
+        let masked = (leadingByte & Asn1MetadataBitmask.CLASS.rawValue) >> 6
+        
+        // Force unwrap is safe due to bitmask
+        self = Asn1IdClass(rawValue: masked)!
+    }
+}
+
+enum Asn1IdMethod: UInt8 {
+    case PRIMITIVE      = 0x00
+    case CONSTRUCTED    = 0x01
+    
+    init(leadingByte: UInt8) {
+        let masked = (leadingByte & Asn1MetadataBitmask.METHOD.rawValue) >> 5
+        
+        // Force unwrap is safe due to bitmask
+        self = Asn1IdMethod(rawValue: masked)!
+    }
+}
+
+enum Asn1IdUniversalTag: UInt8 {
+    case BER_EOC                = 0x00
+    case BER_BOOLEAN            = 0x01
+    case BER_INTEGER            = 0x02
+    case BER_BIT_STR            = 0x03
+    case BER_OCTET_STRING       = 0x04
+    case BER_NULL               = 0x05
+    case BER_OID                = 0x06
+    case BER_OBJ_DESC           = 0x07
+    case BER_EXTERNAL           = 0x08
+    case BER_REAL               = 0x09
+    case BER_ENUM               = 0x0A
+    case BER_EMBED_PDV          = 0x0B
+    case BER_UTF8_STRING        = 0x0C
+    case BER_RELATIVE_OID       = 0x0D
+    case BER_OLD_TIME           = 0x0E
+    case ISO_FUTURE_USAGE       = 0x0F
```

Comment:
Should be omitted.

Reply:
Not omitting because having all values 0x00-0x1F allows force unwrap safely after bitmask is applied.